src/app/carbon-estimation/carbon-estimation.component.ts

```diff
@@ -53,73 +55,23 @@ export class CarbonEstimationComponent implements OnInit, OnDestroy {
     this.resizeSubscription.unsubscribe();
   }
 
+  public onResize(innerHeight: number, innerWidth: number, screenHeight: number): void {
+    this.hasResized = true;
+    this.chartHeight = this.getChartHeight(innerHeight, innerWidth, screenHeight);
+  }
+
   public onExpanded(): void {
     this.changeDetectorRef.detectChanges();
     this.onResize(window.innerHeight, window.innerWidth, window.screen.height);
   }
 
-  onResize(innerHeight: number, innerWidth: number, screenHeight: number): void {
-    const chartHeight = this.getChartHeight(innerHeight, innerWidth, screenHeight);
-    this.chart?.updateOptions({ chart: { height: chartHeight } });
-  }
-
-  private getOverallEmissionPercentages(carbonEstimation: CarbonEstimation): ApexAxisChartSeries {
-    return [
-      {
-        name: `${EmissionsLabels.Upstream} - ${this.getOverallPercentageLabel(carbonEstimation.upstreamEmissions)}`,
-        color: EmissionsColours.Upstream,
-        data: this.getEmissionPercentages(carbonEstimation.upstreamEmissions, EmissionsLabels.Upstream),
-      },
-      {
-        name: `${EmissionsLabels.Direct} - ${this.getOverallPercentageLabel(carbonEstimation.directEmissions)}`,
-        color: EmissionsColours.Direct,
-        data: this.getEmissionPercentages(carbonEstimation.directEmissions, EmissionsLabels.Direct),
-      },
-      {
-        name: `${EmissionsLabels.Indirect} - ${this.getOverallPercentageLabel(carbonEstimation.indirectEmissions)}`,
-        color: EmissionsColours.Indirect,
-        data: this.getEmissionPercentages(carbonEstimation.indirectEmissions, EmissionsLabels.Indirect),
-      },
-      {
-        name: `${EmissionsLabels.Downstream} - ${this.getOverallPercentageLabel(carbonEstimation.downstreamEmissions)}`,
-        color: EmissionsColours.Downstream,
-        data: this.getEmissionPercentages(carbonEstimation.downstreamEmissions, EmissionsLabels.Downstream),
-      },
-    ].filter(entry => entry.data.length !== 0);
-  }
-
-  private getAriaLabel(emission: ApexAxisChartSeries): string {
-    return `Estimation of emissions. ${emission.map(entry => this.getAriaLabelForCategory(entry as ApexChartSeriesItem)).join(' ')}`;
-  }
-
-  private getEmissionAriaLabel(chartData: ApexAxisChartSeries, isPlaceholder: boolean) {
-    return isPlaceholder ? 'Placeholder for estimation of emissions' : this.getAriaLabel(chartData);
-  }
-
-  private getAriaLabelForCategory(series: ApexChartSeriesItem): string {
-    const category = series.name.replace('-', 'are');
-    return `${category}${this.getEmissionMadeUp(series.data)}`;
-  }
-
-  private getEmissionMadeUp(emission: ApexChartDataItem[]): string {
-    if (emission.length === 0) {
-      return '.';
+  public treemapSelected(): void {
+    if (this.hasResized || this.hasEstimationUpdated) {
+      this.hasResized = false;
+      this.hasEstimationUpdated = false;
+      this.changeDetectorRef.detectChanges();
+      this.treemap.chart?.updateOptions({});
     }
```

Comment:
```suggestion
  public treemapSelected(): void {
    this.changeDetectorRef.detectChanges();
    this.treemap.chart?.updateOptions({});
```
Do we need the conditional logic here? Could this be simplified to the above?

Reply:
I would be keen to keep the conditional logic, as it stops unnecessary re-rendering of the treemap

Reply:
Out of interest I did a quick profile of switching between the tabs with and without the conditional:
![image](https://github.com/user-attachments/assets/32133f71-ceff-410a-9ff0-d617206fe75d)
vs
![image](https://github.com/user-attachments/assets/b13d60ad-3580-43d2-8304-21db40250ce5)
I didn't see much difference visibly but it's obviously doing more work than needed.